icmplib/sockets.py

```diff
@@ -146,6 +154,31 @@ def _set_traffic_class(self, traffic_class):
         '''
         raise NotImplementedError
 
+    def _bind_to_device(self, interface):
+        '''
+        Bind the socket to a specific network interface.
+
+        Only available on Linux. Ignored on macOS and Windows.
+
+        '''
+        # Not available on Windows or macOS
+        if PLATFORM_WINDOWS or PLATFORM_MACOS:
+            return
+
+        # Linux uses SO_BINDTODEVICE
+        # SO_BINDTODEVICE requires CAP_NET_RAW capability on Linux
+        # Interface name must be encoded to bytes
+        # Try to get SO_BINDTODEVICE constant, fallback to 25 for compatibility
+        SO_BINDTODEVICE = getattr(socket, "SO_BINDTODEVICE", 25)
+        try:
+            self._sock.setsockopt(
+                socket.SOL_SOCKET,
+                SO_BINDTODEVICE,
+                interface.encode("ascii"),
+            )
+        except (OSError) as err:
+            raise ICMPSocketError(f"Cannot bind to interface {interface}: {err}")
```

Comment:
Nit: `from None` will completely replace the original exception instead of making two chained exceptions.
Chained exceptions will produce tracebacks saying "while handling this, another exception occurred". Since we already include the first exception text in the second exception text, there's no need to chain them.

```suggestion
            raise ICMPSocketError(f"Cannot bind to interface {interface}: {err}") from None
```